### PR TITLE
Finalize javascript_nodejs/10.prompt-validations

### PR DESCRIPTION
**Pull request by benbrown**
- [x] Is numbered and named correctly and following the spec
- [x] All files have been spell checked
- [x] All style cop/ linting issues have been addressed
- [x] Include all DCR changes
- [x] Explicit state save
- [x] Method name changes begin, end etc
- [x] Does not use mainDialog unless that file actually extends Dialog/ ComponentDialog
- [x] Compiles with latest packages
- [x] Runs per instructions in README
- [x] Add commented out code to use blob storage
- [x] Update to look for production/ local (bot configuration)
- [x] onTurnError at adapter
- [x] Include .env or appsettings.json or web.config with botFilePath and botFileSecret
- [x] Deployment scripts folder has been cleaned up and only includes msbot export files
- [x] README to include
   - [x] msbot clone steps
   - [x] Tools install/ uninstall
   - [x] Concise set of steps to run the sample
- [x] ~C# sample -- Include .bot file in the build output~
- [ ] All samples include the latest botfile (no "secretKey" field; replaced with "padLock"). Use latest MSBOT off the tools MyGet feed to generate the bot file
- [x] remove and sort usings
- [x] ~remove azuredeploy.* under deploymentscripts folder (if any)~

### CODE REVIEW

**Comment by benbrown** [comment]
Applied all previous tune-ups to this one!

File: samples/javascript_nodejs/10.prompt-validations/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+botFilePath=prompt-validations-bot.bot
```
Comment (vishwacsena) [inline samples/javascript_nodejs/10.prompt-validations/.env]: Let's add botFileSecret=

File: samples/javascript_nodejs/10.prompt-validations/bot.js [hunk 2]
```diff
@@ -64,7 +64,7 @@ class MainDialog {
             // If no response has been sent, start the onboarding dialog.
             if (!turnContext.responded) {
                 await dc.beginDialog(ONBOARD_USER);
-            } 
+            }
         } else if (
             turnContext.activity.type === ActivityTypes.ConversationUpdate &&
```
Comment (vishwacsena) [inline samples/javascript_nodejs/10.prompt-validations/bot.js]: This needs to be cleaned up. Welcoming users section is not correct. Can you bring this section over from the basic-bot sample?

File: samples/javascript_nodejs/10.prompt-validations/prompts/dobPrompt/index.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const DATE_LOW_BOUNDS = new Date('1918-08-24');
 const DATE_HIGH_BOUNDS = new Date('2018-08-24');
 
 // This is a custom DateTimePrompt that requires the date to be between DATE_LOW_BOUNDS and DATE_HIGH_BOUNDS.
-module.exports = class DOBPrompt extends DateTimePrompt {
+module.exports.DobPrompt = class DobPrompt extends DateTimePrompt {
     constructor(dialogId) {
         super(dialogId, async (prompt) => {
             try {
```
Comment (vishwacsena) [inline samples/javascript_nodejs/10.prompt-validations/prompts/dobPrompt/index.js]: We should add a comment here to point readers to our text-recognizers suite .. 
https://github.com/Microsoft/Recognizers-Text/tree/master/JavaScript/packages/recognizers-text-suite

may be post ignite we can update this to show using the Timex and resolvers with constraints to have more impactful validation rules.. 

See [here](https://github.com/Microsoft/BotBuilder-Samples/blob/vishwac/cafebotv2/javascript_nodejs/50.contoso-cafe-bot/dialogs/shared/stateProperties/reservationProperty.js) for an usecase in cafebot.

Comment (benbrown) [reply samples/javascript_nodejs/10.prompt-validations/prompts/dobPrompt/index.js]: good idea, it is an impressive library

File: samples/javascript_nodejs/10.prompt-validations/bot.js (hunk 2 above)
Comment (benbrown) [reply samples/javascript_nodejs/10.prompt-validations/bot.js]: Not sure what you want me to do to clean it up.  

If we bring the other class in, that's fine, but would effectively replace this sample which demonstrates multiple different type of custom prompt validators, whereas that one only has string validators.

But I'd like to know what you don't like about how I did this too!

Comment (benbrown) [reply samples/javascript_nodejs/10.prompt-validations/bot.js]: Updated!

**Pull request merged**